src/deploy.ts

```diff
@@ -118,6 +120,148 @@ namespace Deploy {
+        const ocOptions = Oc.getOptions({
+            "docker-server": registryServer, "docker-username": registryUsername, "docker-password": registryPassword,
+        });
+        const ocExecArgs = [
+            Oc.Commands.Create, Oc.SubCommands.Secret, "docker-registry", pullSecretName, ...ocOptions,
+        ];
+
+        if (namespaceArg) {
+            ocExecArgs.push(namespaceArg);
+        }
+
+        await Oc.exec(ocExecArgs);
+
+        await addLabelToSecret(pullSecretName, namespaceArg);
+    }
+
+    export async function linkSecretToServiceAccount(pullSecretName: string, namespaceArg?: string): Promise<void> {
+        const defaultServiceAccount = "default";
+        ghCore.info(`🔗 Linking secret "${pullSecretName}" to the service account "${defaultServiceAccount}"...`);
+        const ocOptions = Oc.getOptions({ for: "pull" });
+        const ocExecArgs = [
+            Oc.Commands.Secrets, Oc.SubCommands.Link, defaultServiceAccount, pullSecretName, ...ocOptions,
+        ];
+
+        if (namespaceArg) {
+            ocExecArgs.push(namespaceArg);
+        }
+
+        await Oc.exec(ocExecArgs);
+    }
+
+    async function addLabelToSecret(pullSecretName: string, namespaceArg?: string): Promise<void> {
+        ghCore.info(`Adding label "${secretLabel}" to secret "${pullSecretName}"`);
+        const ocExecArgs = [
+            Oc.Commands.Label, Oc.SubCommands.Secret, pullSecretName, secretLabel,
+        ];
+
+        if (namespaceArg) {
+            ocExecArgs.push(namespaceArg);
+        }
+
+        await Oc.exec(ocExecArgs);
+    }
+
+    export async function isPullSecretExists(pullSecretName: string, namespaceArg?: string): Promise<boolean> {
+        ghCore.info(`🔎 Checking if secret "${pullSecretName}" exists`);
+        const ocExecArgs = [
+            Oc.Commands.Get, Oc.SubCommands.Secret, pullSecretName,
+        ];
+        if (namespaceArg) {
+            ocExecArgs.push(namespaceArg);
+        }
+        try {
+            await Oc.exec(ocExecArgs, { group: true });
+            return true;
+        }
+        catch (error) {
+            ghCore.debug(error);
+        }
+
+        return false;
+    }
+
+    async function checkPullSecretWithLabel(pullSecretName: string, namespaceArg?: string): Promise<boolean> {
+        ghCore.info(`🔎 Checking if secret "${pullSecretName}" with label "${secretLabel}" exists`);
+        const jsonPath = "{.items[*].metadata.name}";
+        const ocOptions = Oc.getOptions({ selector: secretLabel, output: "" });
+
+        const ocExecArgs = [
+            Oc.Commands.Get, Oc.SubCommands.Secret, ...ocOptions, `jsonpath=${jsonPath}{"\\n"}`,
+        ];
+
+        if (namespaceArg) {
+            ocExecArgs.push(namespaceArg);
+        }
+        const execResult = await Oc.exec(ocExecArgs);
+
+        const secretsList = execResult.out.trim().split(" ");
+
+        for (const secret of secretsList) {
```

Comment:
it is the same semantics but i like `Array.some` for this (which in some languages may be called `any`). it returns true if any item matches the condition.

```js
const secretExists = secretsList.some((secretName) => secretName === pullSecretName);
```

Reply:
Thanks! I didn't knew this syntax.